src/dopri.c

```diff
@@ -731,7 +727,7 @@ const double* dopri_find_time(dopri_data *obj, double t) {
     const double
       t0 = ((double*) ring_buffer_tail(obj->history))[idx_t],
       t1 = ((double*) ring_buffer_tail_offset(obj->history, n - 1))[idx_t];
-    idx0 = (t1 - t0) / (n - 1);
+    idx0 = min_size((t - t0) / (t1 - t0) / (n - 1), n - 1);
```

Comment:
This is an unrelated bug fix that turned up debugging the malaria model - the initial interpolation to start the delay search from was completely incorrect. That was ok except it made things a bit slower unless it also out it out of range which caused a crash